### PR TITLE
Added new tuple gate, elevator

### PR DESCRIPTION
**Pull request by Donach**
Added new functions to control gates
Reworked native functions for controlling elevators

### CODE REVIEW

File: wurst/_handles/Destructable.wurst [hunk 1]
```diff
@@ -69,6 +69,81 @@ public function destructable.setOccluderHeight(real height)
...
+		KillDestructable(this.killable)
+
+/** Used to close a destructable gate */
+public function gate.close()
+	this.open()
+	DestructableRestoreLife(this.killable, GetDestructableMaxLife(this.killable), true)
+	SetDestructableAnimation(this.killable, "stand")
+
+/** Used to destroy a destructable gate */
+public function gate.kill()
```
Comment (Cokemonkey11) [inline wurst/_handles/Destructable.wurst]: Any reason this should be public?

File: wurst/_handles/Destructable.wurst [hunk 2]
```diff
@@ -69,6 +69,81 @@ public function destructable.setOccluderHeight(real height)
...
+	this.open()
+	SetDestructableAnimation(this.killable, "death")
+
+/** Used to toggles a destructable gate open close state */
+public function gate.setOpen(bool isOpen)
+	if GetDestructableLife(this.killable) > 0 and not isOpen
```
Comment (Cokemonkey11) [inline wurst/_handles/Destructable.wurst]: Why is this check needed? What badness happens if you do `isOpen ? this.open() : this.close()`?

I can imagine what might go wrong, but it's not clear. Needs comments answering `why?`

File: wurst/_handles/Destructable.wurst [hunk 3]
```diff
@@ -69,6 +69,81 @@ public function destructable.setOccluderHeight(real height)
...
+		this.close()
+	else if isOpen
+		this.open()
+
+tuple elevator(destructable elevatable)
+constant let ELEVATOR_HEIGHT_MIN = 1
```
Comment (Cokemonkey11) [inline wurst/_handles/Destructable.wurst]: it's inconsequential, but FYSA, `let` here isn't necessary

File: wurst/_handles/Destructable.wurst (hunk 2 above)
Comment (Donach) [reply wurst/_handles/Destructable.wurst]: No reason really, I guess I took it from the original BJ functions where it was implemented like this.

File: wurst/_handles/Destructable.wurst (hunk 1 above)
Comment (Donach) [reply wurst/_handles/Destructable.wurst]: Yes, the gate has "death" and "death alternate" (aka open) animation.
Kill triggers the "destroy" animation, while the regular open triggers the alternative animation. Different use cases.

**Comment by Frotty** [comment]
Looks good, but you should fix the hotdocs.
"Used to ..." doesn't really make sense, just describe what it does. "Opens a destructable gate".
Also the one for "setOpen" is wrong, it doesn't "toggle". Toggle implies it would open if closed, and close of opened, not set by a flag.

**Comment by Frotty** [comment]
It looks like https://github.com/wurstscript/WurstStdlib2/pull/243 still contains these changes.
@Donach please separate and let's get this merged!

File: wurst/_handles/Destructable.wurst [hunk 4]
```diff
@@ -69,6 +69,81 @@ public function destructable.setOccluderHeight(real height)
...
+public function elevator.elevatorSetHeight(int height)
+	this.setHeight(height)
+
+/** Used to get a destructable elevator height */
+public function elevator.getHeight() returns int
+	let height = ELEVATOR_HEIGHT_MIN + (this.elevatable.getOccluderHeight() / bj_CLIFFHEIGHT).toInt()
+	return height < ELEVATOR_HEIGHT_MIN or height > ELEVATOR_HEIGHT_MAX ? ELEVATOR_HEIGHT_MIN : height
+
+/** Returns elevator adjusted height based on given bounds */
+function isElevatorHeightInBounds(int height) returns bool
```
Comment (Frotty) [inline wurst/_handles/Destructable.wurst]: unused private function? should be public? The one below too maybe?

Comment (Donach) [reply wurst/_handles/Destructable.wurst]: fixed

**Comment by Frotty** [comment]
Nice, thanks 🍭

**Pull request merged**